luigi/contrib/sqla.py

```diff
@@ -153,22 +155,37 @@ class SQLAlchemyTarget(luigi.Target):
     to create a task to write to the database.
     """
     marker_table = None
+    _engine = None  # sqlalchemy engine
+    _pid = None  # the pid of the sqlalchemy engine object
 
     def __init__(self, connection_string, target_table, update_id, echo=False):
         """
         Constructor for the SQLAlchemyTarget.
 
-        :param connection_string: (str) SQLAlchemy connection string
-        :param target_table: (str) The table name for the data
-        :param update_id: (str) An identifier for this data set
-        :param echo: (bool) Flag to setup SQLAlchemy logging
+        :param connection_string: SQLAlchemy connection string
+        :type connection_string: str
+        :param target_table: The table name for the data
+        :type target_table: str
+        :param update_id: An identifier for this data set
+        :type update_id: str
+        :param echo: Flag to setup SQLAlchemy logging
+        :type echo: bool
         :return:
         """
         self.target_table = target_table
         self.update_id = update_id
-        self.engine = sqlalchemy.create_engine(connection_string, echo=echo)
+        self.connection_string = connection_string
+        self.echo = echo
         self.marker_table_bound = None
 
+    @property

```

Comment:
The engine instance is checked to make sure it belongs to the same process as that of the worker process.